jsonnet/components/prometheus-user-workload.libsonnet

```diff
@@ -464,6 +464,9 @@ function(params)
               },
             },
           },
+          {
+            name: 'prometheus',
+          },
```

Comment:
I think we should add a comment that explains why an empty container with name `prometheus` is needed.

Reply:
We missed the comment on 4.14 branch . Should we add it from there?

Reply:
I think a comment would be nice since an empty container with `prometheus` doesn't tell the reader why this is important. I.E. why this is required for the startup probes to be setup properly.

Reply:
Added comment on source file https://github.com/openshift/cluster-monitoring-operator/pull/1943